src/bindings/swift/gen_swift.rs

```diff
@@ -0,0 +1,498 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::Result;
+use askama::Template;
+
+use crate::interface::*;
+
+// Some config options for it the caller wants to customize the generated python.
+// Note that this can only be used to control details of the python *that do not affect the underlying component*,
+// sine the details of the underlying component are entirely determined by the `ComponentInterface`.
+pub struct Config {
+    // No config options yet.
+}
+
+impl Config {
+    pub fn from(_ci: &ComponentInterface) -> Self {
+        Config {
+            // No config options yet
+        }
+    }
+}
+
+#[derive(Template)]
+#[template(
+    ext = "h",
+    escape = "none",
+    source = r#"
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef struct RustBuffer {
+    int64_t len;
+    uint8_t *_Nullable data;
+} RustBuffer;
+
+{% for func in ci.iter_ffi_function_definitions() -%}
+    {%- match func.return_type() -%}{%- when Some with (type_) %}{{ type_|decl_c }}{% when None %}void{% endmatch %} {{ func.name() }}(
+      {%- for arg in func.arguments() %}
+      {{ arg.name() }}: {{ arg.type_()|decl_c }}{% if loop.last %}{% else %},{% endif %}
+      {%- endfor %}
+      // TODO: When we implement error handling, there will be a `*_Nonnull out_err` param here.
+    );
+{% endfor -%}
+"#
+)]
+pub struct BridgingHeader<'config, 'ci> {
+    _config: &'config Config,
+    ci: &'ci ComponentInterface,
+}
+
+impl<'config, 'ci> BridgingHeader<'config, 'ci> {
+    pub fn new(config: &'config Config, ci: &'ci ComponentInterface) -> Self {
+        Self {
+            _config: config,
+            ci,
+        }
+    }
+}
+
+#[derive(Template)]
+#[template(
+    ext = "swift",
+    escape = "none",
+    source = r#"
+// This file was autogenerated by some hot garbage in the `uniffi` crate.
+// Trust me, you don't want to mess with it!
+
+extension Data {
+    init(rustBuffer: RustBuffer) {
+        // TODO: This copies the buffer. Can we read directly from a
+        // Rust buffer?
```

Comment:
Copying on the way in and out simplifies things—I mainly did that to get it working, but we should see if we can refactor to avoid it.

Reply:
+1 to "get it working" FWIW!